frontend/src/core/plural/components/PluralSelector.test.js

```diff
@@ -78,20 +78,28 @@ describe('<PluralSelector>', () => {
             },
             router: {
                 location: {
-                    pathname: '/kg/pro/all/',
+                    pathname: '/kg/firefox/all-resources/',
                     search: '?string=42',
                 },
+                // `action` is required because
+                // https://github.com/supasate/connected-react-router/issues/312#issuecomment-500968504
+                // Please note the initial `LOCATION_CHANGE` action can and must
+                // be supressed via the `noInitialPop` prop in
+                // `ConnectedRouter`, otherwise it'll cause side-effects like
+                // executing reducers and hence altering initial state values.
+                action: 'some-string-to-please-connected-react-router',
             },
         };
         const store = createReduxStore(initialState);
         const dispatchSpy = sinon.spy(store, 'dispatch');
 
-        const wrapper = shallowUntilTarget(
-            <PluralSelector store={store} resetEditor={sinon.spy()} />,
-            PluralSelectorBase,
-        );
+        const root = mountComponentWithStore(PluralSelector, store, {
+            resetEditor: sinon.spy(),
+        });
+        const wrapper = root.find(PluralSelectorBase);
 
-        wrapper.find('button').first().simulate('click');
+        const button = wrapper.find('button').first();
+        button.prop('onClick')();
```

Comment:
I really struggled with this one. Apparently `.simulate()` doesn't quite work when components are mounted (as opposed to shallow-rendered). If you read enzyme GitHub issues, the maintainer strongly suggests to stay away from `.simulate()` because it doesn't do what it appears to be doing, and instead suggests to directly invoke the component's handler function.

Reply:
Sure, done.